### PR TITLE
Add DOCS for promise changes and update all features and tests to cater for it

### PR DESCRIPTION
**Pull request by ScopeyNZ**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -20,13 +20,32 @@ https://commercejs.com
...
+<script type="text/javascript" src="https://cdn.chec.io/v2/commerce.js"></script>
 ```
 
 ## Documentation
 
 See the [documentation webpage](https://commercejs.com/docs).
 
+## Upgrading
+
+The major change in Commerce.js v2 is that most methods now return a promise that is furfilled when the http request is
```
Comment (robbieaverill) [inline README.md]:
```suggestion
The major change in Commerce.js v2 is that most methods now return a [promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) that is fulfilled when the HTTP request is
```

File: README.md [hunk 2]
```diff
@@ -20,13 +20,32 @@ https://commercejs.com
...
+  // ...
+});
+```
+
+Becomes
+
+```js
+Commerce.Cart.retrieve().then(function (data) {
+  // ...
+})
```
Comment (robbieaverill) [inline README.md]: You could use a diff block here:

```diff
- Commerce.Cart.retrieve(function (data) {
+ Commerce.Cart.retrieve().then(function (data) {
  // ...
});
```

File: src/features/checkout.js [hunk 3]
```diff
@@ -12,10 +12,7 @@ class Checkout {
    * @param {string} token
    */
   protect(token) {
-    this.commerce.request(
-      `checkouts/${token}/protect`,
-      'GET',
-      null,
+    this.commerce.request(`checkouts/${token}/protect`).then(
```
Comment (robbieaverill) [inline src/features/checkout.js]: You're returning from the other methods, does this need one too?

Comment (ScopeyNZ) [reply src/features/checkout.js]: I didn't think it was necessary... I suppose it doesn't hurt though...

**Pull request merged**